timer/component.go

```diff
@@ -0,0 +1,34 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+package timer
+
+import "sync/atomic"
+
+type component struct {
+	// name through which the component is identified
+	name string
+	// Updated each time a run of a sub-timer ends
+	spent atomic.Int64
+}
+
+// components store the data shared between child timers of the same component name
+type components struct {
+	lookup  map[string]*component
+	storage []component
+}
```

Comment:
(Micro-optimization, optional) I'm actually wondering if the GC would have less work to do if, instead of having `lookup map[string]*component` it was `lookup map[string]int` where the value is the index of the component in the `storage` map. That'd make fewer pointers for the GC to have to keep track of to determine reachability.

Reply:
I tried what you are describing along a few other options (like only using a map without a slice) and this is the best one right now. Note that all the benchmarks spend more than 90% of their time in the `time` package anyway...